infomapcog/distances.py

```diff
@@ -73,32 +73,57 @@ def needleman_wunsch(x, y, lodict={}, gop=-2.5, gep=-1.75):
     'lodict' must be a dictionary with all symbol pairs as keys and
     match scores as values, or a False value (including an empty
     dictionary) to denote (-1, 1) scores. gop and gep are gap
-    penalties for opening/extending a gap.
+    penalties for opening/extending a gap; alternatively, you can set
+    'gop' to None and provide element/gap alignment costs.
 
     Returns the alignment score and one optimal alignment.
+    
+    >>> needleman_wunsch("AAAAABBBB", "AACAABBCB")
+    (5.0, [('A', 'A'), ('A', 'A'), ('A', 'C'), ('A', 'A'), ('A', 'A'), ('B', 'B'), ('B', 'B'), ('B', 'C'), ('B', 'B')])
+    >>> needleman_wunsch("banana", "mancala", local=True)
```

Comment:
sw_align is there, nw_align is there, semi-global alignment is there in form of sound-class-based alignments with prosodic strings, so it will be difficult to simply pull that. But all code is flat, not using any numpy, which you also do not want to use if you make a thousand of those pairwise alignments in a run. I suggest you first have a quick look at nw_align and sw_align in lingpy.algorithm.cython._malign.py (but this is automatically converted from the pyx version for cython, yet should be self-explaining).